### PR TITLE
Add YOLO validation script

### PR DESCRIPTION
**Pull request by rahul-tuli**
This PR adds a `val_onnx.py` script to the `object_detection` module. This script can be use to validate YOLO `onnx models`/`SparseZoo` stubs directly on COCO

This script borrows heavily from the original YOLO validation script from [here](https://github.com/neuralmagic/sparseml/blob/main/src/sparseml/pytorch/object_detection/val.py)

The script was tested with the following command:

Noting the following stub is used by default `"zoo:cv/detection/yolov5-l/pytorch/ultralytics/coco/pruned-aggressive_98"`
```bash
sparseml.object_detection.eval_onnx \
    --data coco.yaml --batch-size 1
```

The results are as follows:

```bash
DeepSparse Engine, Copyright 2021-present / Neuralmagic, Inc. version: 0.13.0 (82a6773e) (optimized) (system=avx512, binary=avx512)
val: Scanning '/home/rahul/virtual_environments/sparseml3.8/lib/python3.8/site-p
               Class     Images     Labels          P          R     mAP@.5 mAP@

                 all       5000      36335      0.718      0.591      0.638      0.444
Speed: 0.0ms pre-process, 123.1ms inference, 0.0ms NMS per image at shape (1, 3, 640, 640)

Evaluating pycocotools mAP... saving runs/val/exp36/_predictions.json...
loading annotations into memory...
Done (t=0.62s)
creating index...
index created!
Loading and preparing results...
DONE (t=3.23s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=38.90s).
Accumulating evaluation results...
DONE (t=5.99s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.450
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.640
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.483
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.271
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.502
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.585
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.346
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.557
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.603
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.407
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.654
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.742
Results saved to runs/val/exp36

Process finished with exit code 0

```

### CODE REVIEW

File: src/sparseml/pytorch/object_detection/val_onnx.py [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+from sparseml.onnx.utils import get_tensor_dim_shape
+
+
+try:
+    from deepsparse import Pipeline
+
+except Exception as deepsparse_error:
+    raise RuntimeError(
+        "Unable to import Pipeline from deepsparse"
+        f"DeepSparse>=12.1 must be installed {deepsparse_error}"
```
Comment (bfineran) [inline src/sparseml/pytorch/object_detection/val_onnx.py:38]: * now deepsparse 13.0

Comment (rahul-tuli) [reply src/sparseml/pytorch/object_detection/val_onnx.py:38]: Thanks, Updated!

**Comment by bfineran** [comment]
noting we need to look into accuracy issues here @rahul-tuli

**Comment by rahul-tuli** [comment]
> noting we need to look into accuracy issues here @rahul-tuli
Resolved!

**Comment by rahul-tuli** [comment]
Noting this PR is being closed in favor of migrating this script to yolov5 fork,

The new functionality has been added to the following PRs instead:
- #824 
- [YOLOv5 PR #52](https://github.com/neuralmagic/yolov5/pull/52)

**Pull request closed without merging**